tests/test_requests.py

```diff
@@ -526,6 +526,19 @@ def test_proxy_error(self):
         with pytest.raises(ProxyError):
             requests.get('http://localhost:1', proxies={'http': 'non-resolvable-address'})
 
+    def test_proxy_error_on_bad_url(self):
+        with pytest.raises(ProxyError):
+            requests.get('https://github.com', proxies={'https': 'http:/badproxyurl:3128'})
```

Comment:
@nehaljwani we prefer to avoid making requests live servers from the test suite. Can you please use the httpbin fixture like the other tests? Thanks!

Reply:
+1 to @nateprewitt's request

Reply:
Okay, I've updated the tests with httpbin